src/rules/no-invalid-properties.js

```diff
@@ -124,6 +124,127 @@ export default {
+		 * @param {string[]} valueList Array to collect processed values
+		 * @param {Map<string,CssLocationRange>} valuesWithVarLocs Map of values to their locations
+		 * @returns {boolean} Whether processing was successful
+		 */
+		function processVarFunction(varNode, valueList, valuesWithVarLocs) {
+			const varValue = vars.get(varNode.children[0].name);
+
+			if (varValue) {
+				const varValueText = sourceCode.getText(varValue).trim();
+				valueList.push(varValueText);
+				valuesWithVarLocs.set(varValueText, varNode.loc);
+				return true;
+			}
+
+			// If the variable is not found and doesn't have a fallback value, report it
+			if (varNode.children.length === 1) {
+				if (!allowUnknownVariables) {
+					context.report({
+						loc: varNode.children[0].loc,
+						messageId: "unknownVar",
+						data: { var: varNode.children[0].name },
+					});
+					return false;
+				}
+				return true;
+			}
+
+			// Handle fallback values
+			if (varNode.children[2].type !== "Raw") {
+				return true;
+			}
+
+			const fallbackVarList = getVarFallbackList(
+				varNode.children[2].value.trim(),
+			);
+
+			if (fallbackVarList.length === 0) {
+				const fallbackValue = varNode.children[2].value.trim();
+				valueList.push(fallbackValue);
+				valuesWithVarLocs.set(fallbackValue, varNode.loc);
+				return true;
+			}
+
+			// Process fallback variables
+			for (const fallbackVar of fallbackVarList) {
+				if (fallbackVar.startsWith("--")) {
+					const fallbackVarValue = vars.get(fallbackVar);
+					if (fallbackVarValue) {
+						const fallbackValue = sourceCode
+							.getText(fallbackVarValue)
+							.trim();
+						valueList.push(fallbackValue);
+						valuesWithVarLocs.set(fallbackValue, varNode.loc);
+						return true;
+					}
+				} else {
+					valueList.push(fallbackVar.trim());
+					valuesWithVarLocs.set(fallbackVar.trim(), varNode.loc);
+					return true;
+				}
+			}
+
+			// No valid fallback found
+			if (!allowUnknownVariables) {
+				context.report({
+					loc: varNode.children[0].loc,
+					messageId: "unknownVar",
+					data: { var: varNode.children[0].name },
+				});
+				return false;
+			}
+
+			return true;
+		}
+
+		/**
+		 * Process a nested function by recursively handling its children
+		 * @param {FunctionNodePlain} funcNode The function node
+		 * @param {Map<string,CssLocationRange>} valuesWithVarLocs Map of values to their locations
+		 * @returns {string|null} The processed function string or null if processing failed
+		 */
+		function processNestedFunction(funcNode, valuesWithVarLocs) {
+			const nestedValueList = [];
+
+			for (const nestedChild of funcNode.children) {
+				if (
+					nestedChild.type === "Function" &&
+					nestedChild.name === "var"
+				) {
+					if (
+						!processVarFunction(
+							nestedChild,
+							nestedValueList,
+							valuesWithVarLocs,
+						)
+					) {
+						return null;
+					}
+				} else if (nestedChild.type === "Function") {
+					// Recursively process nested functions
```

Comment:
Here we're doing a traversal on top of the traversal that is already happening in the rule, which is inefficient.

A more efficient approach would be to use the `Function` visitor method to do the traversal automatically. Can you take a look at that?